jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -7301,6 +7318,21 @@ public boolean matches(String expr) {
 		return Pattern.matches(expr, this);
 	}
 
+	/*
```

Comment:
I think we can have one copy of common code, i.e. #1931. Please move the helper so its shared between both versions.

Reply:
Shared the helper method with minimum modification to existing duplication. 
Will open separated PR to move other shared field/methods.
Please review.